packages/mediapipe-core/lib/src/ffi_utils.dart

```diff
@@ -76,3 +76,60 @@ int _length(Pointer<Uint8> codeUnits) {
   }
   return length;
 }
+
+/// Converts a pointer to a (representing a list of) floats to a list of
+/// Dart doubles.
+List<double> toDartListDouble(Pointer<Float> floats, {int? length}) {
+  if (floats.isNullPointer) {
+    throw Exception('Unexpected nullptr passed to `toDartListDouble`.');
+  }
+  final codeUnits = floats.cast<Float>();
+  if (length != null) {
+    RangeError.checkNotNegative(length, 'length');
+  } else {
+    length = lengthFloats(codeUnits);
+  }
+  final value = codeUnits.asTypedList(length);
+  return value;
```

Comment:
Given the similarity in `toUint8List` can we wrap some of this in a method to reuse?

Reply:
Fully refactoring this code.